@nrwl/angular:webpack-dev-server executor fails with local imports in webpack config

<!-- Please do your best to fill out all of the sections below! -->

## Current Behavior
<!-- What is the behavior that currently you experience? -->
After the cleanup story https://github.com/nrwl/nx/pull/12924 ([this change](https://github.com/nrwl/nx/pull/12924/files#diff-39a45d5a12bd4b48ebc0871fab97966eab51e70137d4e1d48df18ee5171b5da2R114-R118)), custom webpack files lost the possibility to import local libraries. Each serve command ends up with error being thrown regarding unresolved dependencies while build works fine.

## Expected Behavior
<!-- What is the behavior that you expect to happen? -->
<!-- Is this a regression? .i.e Did this used to be the behavior at one point?  -->
The serve command works in the same way as build and allow referencing local libraries.

## Steps to Reproduce
<!-- Help us help you by making it easy for us to reproduce your issue! -->

<!-- Can you reproduce this on https://github.com/nrwl/nx-examples? -->
<!-- If so, open a PR with your changes and link it below. -->
<!-- If not, please provide a minimal Github repo -->
<!-- At the very least, provide as much detail as possible to help us reproduce the issue -->
https://github.com/nrwl/nx-examples/pull/245

1. run `yarn nx build products` -> everything is fine
2. run `yarn nx serve products` -> build fails 

### Failure Logs
<!-- Please include any relevant log snippets or files here. -->
```
⠋ Generating browser application bundles (phase: setup)...
 >  NX   Cannot find module '@nx-example/shared/jsxify'

   Require stack:
   - /home/michal/Documents/projects/nx-examples/apps/products/webpack.config.ts
   - /home/michal/Documents/projects/nx-examples/node_modules/@nrwl/angular/src/builders/utilities/webpack.js
   - /home/michal/Documents/projects/nx-examples/node_modules/@nrwl/angular/src/builders/webpack-dev-server/webpack-dev-server.impl.js
   - /home/michal/Documents/projects/nx-examples/node_modules/@angular-devkit/architect/node/node-modules-architect-host.js
   - /home/michal/Documents/projects/nx-examples/node_modules/@angular-devkit/architect/node/index.js
   - /home/michal/Documents/projects/nx-examples/node_modules/nx/src/adapter/ngcli-adapter.js
   - /home/michal/Documents/projects/nx-examples/node_modules/nx/src/command-line/run.js
   - /home/michal/Documents/projects/nx-examples/node_modules/nx/bin/run-executor.js
   Pass --verbose to see the stacktrace.


 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Running target "products:serve" failed

   Failed tasks:
   
   - products:serve:production
   
   Hint: run the command with --verbose for more details.

   View structured, searchable error logs at https://nx.app/runs/FdSipZJsKE

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Environment
<!-- It's important for us to know the context in which you experience this behavior! -->
<!-- Please paste the result of `nx report` below! -->
```
yarn run v1.22.18
$ nx report

 >  NX   Report complete - copy this into the issue template

   Node : 16.14.2
   OS   : linux x64
   yarn : 1.22.18
   
   nx : 15.1.0-beta.2
   @nrwl/angular : 15.1.0-beta.2
   @nrwl/cypress : 15.1.0-beta.2
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.1.0-beta.2
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.1.0-beta.2
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.1.0-beta.2
   @nrwl/js : 15.1.0-beta.2
   @nrwl/linter : 15.1.0-beta.2
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 15.0.2
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.1.0-beta.2
   @nrwl/react-native : Not Found
   @nrwl/rollup : 15.1.0-beta.2
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.1.0-beta.2
   @nrwl/web : 15.1.0-beta.2
   @nrwl/webpack : 15.1.0-beta.2
   @nrwl/workspace : 15.1.0-beta.2
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
         @ngrx/component-store: 14.0.2
         @ngrx/effects: 14.0.2
         @ngrx/entity: 14.0.2
         @ngrx/router-store: 14.0.2
         @ngrx/store: 14.0.2
         @ngrx/store-devtools: 14.0.2

Done in 0.26s.
```

--- a/packages/angular/src/builders/webpack-dev-server/webpack-dev-server.impl.ts
+++ b/packages/angular/src/builders/webpack-dev-server/webpack-dev-server.impl.ts
@@ -84,6 +84,12 @@ export function executeWebpackDevServerBuilder(
       options.tsConfig = tsConfigPath;
       return options;
     };
+
+    // The buildTargetConfiguration also needs to use the generated tsconfig path
+    // otherwise the build will fail if customWebpack function/file is referencing
+    // local libs. This synchronize the behavior with webpack-browser and
+    // webpack-server implementation.
+    buildTargetConfiguration.tsConfig = tsConfigPath;
   }
 
   return executeDevServerBuilder(options as DevServerBuilderOptions, context, {

